_layouts/py1-lesson.html

```diff
@@ -2,75 +2,65 @@
 layout: default
 fw: true
 ---
-<div class="py1-lesson-page-wrapper">
-    <div class="py1-lesson-sidebar">
-        <div class="py1-section1">
-            <h1 class="py1-lesson-header"> {{ page.num }}. {{ page.title }} </h1>
-            <p class="py1-lesson-page-description">
+<div class="py1-wrapper">
+    <div class="py1-sidebar">
+        <div>
+            <h1 class="py1-lesson-header">
+                {{ page.num }}. {{ page.title }}
+            </h1>
+            <p>
                 {{ content }}
             </p>
 
-            <p class="py1-lesson-page-agenda">
-                <b> agenda </b>
-                {{ page.agenda }}
-            </p>
-
-            <div class="py1-lesson-page-action-bar">
-                {% if page.slides_link %}
-                <a href="{{ page.slides_link }}" target="_blank">slides</a>
-                {% endif %}
-
-                {% if page.worksheet_link %}
-                <a href="{{ page.worksheet_link }}" target="_blank">worksheet</a>
-                {% endif %}
+            {% include py1-lesson-card-content.html %}
 
-                <span class="py1-lesson-duration">{{ page.length }} min</span>
-            </div>
         </div>
 
         {% if page.next_lesson %}
         <div class="py1-lesson-page-next">
             <a href="{{site.baseurl}}/classes/py1/{{page.next_link}}">next lesson: {{ page.next_lesson }}</a>
         </div>
         {% endif %}
+        <div class="py1-lesson-page-home">
+            <a href="{{site.baseurl}}/classes/py1/">back to curriculum home</a>
+        </div>
 
     </div>
 
-    <div class="py1-lesson-page-divider"></div>
+    <div class="py1-divider"></div>
 
     <div class="py1-lesson-content">
-        <!--TODO: make responsive-->
         <iframe width="960" height="450" src={{ page.video_link }} frameborder="0"
             allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
             allowfullscreen></iframe>
 
-        {% if page.p1 or page.p2 or page.p3 %}
+        {% if page.practice %}
         <h1>more practice</h1>
-        {% endif %}
 
-        {% if page.p1 %}
+        {% for item in page.practice %}
         <div class="py1-lesson-practice">
-            <b>question #1</b> <br> <br>
-            {{ page.p1.question }} <br> <br>
-            <a href="{{page.p1.solution}}" target="_blank">solution</a>
+            <b>question #{{ item.problem.num }}</b> <br>
+            {{ item.problem.question | markdownify }} <br>
+            <button onclick="revealSolution({{ item.problem.num }} - 1)">solution</button> <!-- annoying error >:( -->
```

Comment:
Haha, what's the annoying error?

Reply:
<img width="1304" alt="Screen Shot 2020-11-30 at 8 42 00 PM" src="https://user-images.githubusercontent.com/40647794/100698170-ae50e400-334c-11eb-861b-3bd5d9c4819b.png">

VSCode really doesn't like when you embed liquid syntax within js... Oh well it still compiles!